Sources/SwiftInspectorVisitors/TypeSyntaxVisitor.swift

```diff
@@ -79,24 +79,23 @@ public final class TypeSyntaxVisitor: SyntaxVisitor {
   private let typeName: String
 
   private func processNode<Node>(_ node: Node, withName name: String, members: MemberDeclListSyntax) where Node: SyntaxProtocol {
-    var result = [TypeProperties.PropertyData]()
-    let propertyVisitor = PropertySyntaxVisitor(typeName: name) { info in
-      result.append(info)
-    }
+    let propertyVisitor = PropertySyntaxVisitor(typeName: name)
     propertyVisitor.walk(node)
-    onNodeVisit(.init(name: name, properties: result))
+    onNodeVisit(.init(name: name, properties: propertyVisitor.propertiesData))
   }
 }
 
 // MARK: - PropertySyntaxVisitor
 
 private final class PropertySyntaxVisitor: SyntaxVisitor {
 
-  init(typeName: String, onNodeVisit: @escaping (_ info: TypeProperties.PropertyData) -> Void) {
+  init(typeName: String) {
     self.typeName = typeName
-    self.onNodeVisit = onNodeVisit
   }
 
+  /// Information about each of the properties found on the type.
+  private(set) var propertiesData = [TypeProperties.PropertyData]()
```

Comment:
Now we have a property rather than a callback. This seems to be the modern approach.

Reply:
at least that's how I liked to write the visitors. I approve of this method 🙂

Reply:
Yeah, I've come back and forth on this honestly. I liked that the Visitors were fairly decoupled originally and was hoping we could use composition at some point to reuse them without having them know about the types (and having someone that calls into them know instead), but I really like the approach we took of using composition _while having these visitors know about the types they're generating_ I think it makes it really powerful.

Reply:
Just gonna say I'm super happy with how easy https://github.com/fdiaz/SwiftInspector/pull/88 turned out to be. We're developing quite the toolset.

Reply:
I don't think we should _never_ write a "generalized" visitor. For example, regarding https://github.com/fdiaz/SwiftInspector/pull/87#discussion_r610324857, I think we eventually want to have a visitor that looks for a fully qualified type name. And then we'd have a more specific visitor inspect the properties of the types once they are found. But I don't think every visitor needs to be generalized.